packages/api-client-core/src/InternalModelManager.ts

```diff
@@ -43,23 +43,108 @@ const internalHydrationPlan = (modelApiIdentifer: string) => `
 export const internalFindOneQuery = (apiIdentifier: string) => {
   const capitalizedApiIdentifier = capitalizeIdentifier(apiIdentifier);
   return `
-    query InternalFind${capitalizedApiIdentifier}($id: GadgetID!) {
+    query InternalFind${capitalizedApiIdentifier}($id: GadgetID!, $select: [String!]) {
       ${internalHydrationPlan(apiIdentifier)}
       internal {
-        ${apiIdentifier}(id: $id)
+        ${apiIdentifier}(id: $id, select: $select)
       }
     }
     `;
 };
 
-export const internalFindFirstQuery = (apiIdentifier: string, options?: RecordData) => {
-  const capitalizedApiIdentifier = capitalizeIdentifier(apiIdentifier);
+/**
+ * A list of fields to select from the internal API
+ * Matches the format of the Public API `select` option, but only allows going one level deep -- no relationships can be selected using the internal API.
+ *
+ * Supports passing a list of strings as a shorthand.
+ *
+ * @example
+ * { fieldA: true, fieldB: true, fieldC: false }
+ *
+ * @example
+ * ['fieldA', 'fieldB']
+ */
+export type InternalFieldSelection = string[] | { [field: string]: boolean | null | undefined };
+
+/** Options for the api functions that return one record on an InternalModelManager */
+export interface InternalFindOneOptions {
+  /**
+   * What fields to retrieve from the API for this API call
+   **/
+  select?: InternalFieldSelection;
+}
+
+/** Options for functions that query a list of records on an InternalModelManager */
+export interface InternalFindListOptions {
+  /**
+   * A string to search for within all the stringlike fields of the records
+   * Matches the behavior of the Public API `search` option
+   **/
+  search?: string;
+  /**
+   * How to sort the returned records
+   * Matches the format and behavior of the Public API `sort` option
+   *
+   * @example
+   * {
+   *   sort: { publishedAt: "Descending" }
+   * }
+   **/
+  sort?: Record<string, "Ascending" | "Descending"> | Record<string, "Ascending" | "Descending">[];
```

Comment:
In your example, it'd be a,b because since ES6 key iteration order on objects is actually stable and defined, its in the order you see them in the code. This is subtle though, and the GraphQL API actually supports both with a weird coercion thing, so the type supports both here too.